packages/ra-core/src/auth/useLogout.ts

```diff
@@ -59,40 +46,50 @@ const useLogout = (): Logout => {
                 const redirectToParts = (
                     redirectToFromProvider || redirectTo
                 ).split('?');
-                const newLocation: LocationDescriptorObject = {
+                const newLocation: Partial<Path> = {
                     pathname: redirectToParts[0],
                 };
+                let newLocationOptions = {};
+
                 if (
                     redirectToCurrentLocationAfterLogin &&
-                    history.location &&
-                    history.location.pathname
+                    location &&
+                    location.pathname
                 ) {
-                    newLocation.state = {
-                        nextPathname: history.location.pathname,
-                        nextSearch: history.location.search,
+                    newLocationOptions = {
```

Comment:
There can be other options in the location. You're erasing them. setting only the state didn't have this drawback.

Reply:
I don't get this. You mean `replace` and others? The `navigate` function signature isn't the same as the old history push